### ISSUE TITLE
Adapt PanelTimeSeries Split for use with sklearn

### ISSUE THREAD

**Issue by rsueppel** [macrosynergy/macrosynergy#984]
What we need as basic functionality is test of various methods by cross-validation and a set criterion, i.e.

we pass
> the feature and target panels
> a dictionary of classifier or regression methods
> a dictionary of scoring methods based on predicted and true values (acuracy, R2 of custom like Sharpe ratio)  

the method returns
> a dataframe with the scores

for illustration see below convenience function which does a similar thing except that it hard codes the scores and just prints them.

One of two paths must be chosen:

- [x] We tweak PanelTimeSeriesSPlit such that it works with standard sklearn cross-validation methods and processes.
- [x] We build our own classes and methods that apply panel splits 

The former would be preferable.

**Comment by rsueppel** [comment]
Example

def binary_cv_scores(
    dict_clf, splitter, X, y, visualize_split=False, scorings="all", split_avg=False
):
    """ "
    Display cross-validation scores for standard binary (-1/-1) classification and a
    dictionary of alternative classifiers.

    :param <dict> dict_clf: dictionary of classifiers, with values specificying the classification model
        and keys providing suitable labels.
    :param <tuple> splitter: PanelTimeSeriesSplit object that serves the splitting of
        panel data into panel training and test sets.
    :param <pd.DataFrame> X: DataFrame of features.
    :param <pd.Series> y: Series of targets.
    :param <list> scorings: list of scoring strings to be used in cross-validation.
        Default is all. NOT IMPLEMENTED YET.
    :param <bool> split_avg: if True, the cross-validation scores are averaged across
        the splits. Default is False, i.e. predicted and true values are first
        collected and then the scores are calculated. NOT IMPLEMENTED YET.
    """

    split = splitter.split(X, y)

    if visualize_split:
        splitter.visualise_splits(X, y)

    dict_scoring = {
        "Accuracy": accuracy_score,
        "Balanced": balanced_accuracy_score,
        "Precision": lambda y_true, y_pred: precision_score(
            y_true, y_pred, pos_label=1
        ),
        "Sensitivity": lambda y_true, y_pred: recall_score(y_true, y_pred, pos_label=1),
        "F1 score": f1_score,
        "NPV": lambda y_true, y_pred: precision_score(y_true, y_pred, pos_label=-1),
        "Specificity": lambda y_true, y_pred: recall_score(
            y_true, y_pred, pos_label=-1
        ),
        "AUC-ROC": roc_auc_score,
    }

    for clf_name, clf in dict_clf.items():
        print(f"\n{clf_name}\n")

        preds = []
        actuals = []

        for idx, (train_idxs, test_idxs) in enumerate(split):
            # form train and test sets
            X_train, X_test = X.loc[train_idxs], X.loc[test_idxs]
            y_train, y_test = y.loc[train_idxs], y.loc[test_idxs]

            # train model
            tmod = clf.fit(X_train, y_train)

            # predict and collect
            y_pred = tmod.predict(X_test)
            preds.extend(y_pred)
            actuals.extend(y_test)

        rpts = sum(np.array(actuals) == 1) / len(actuals)
        rpcs = sum(np.array(preds) == 1) / len(preds)

        print(f"Ratio of positive test targets: {rpts: .3f}")
        print(f"Ratio of positive classifications: {rpcs: .3f}")

        for k, v in dict_scoring.items():
            print(f"{k}: {v(actuals, preds): .3f}")
